### ISSUE TITLE
doubleMean aggregator not available

### ISSUE THREAD

**Issue by BeanNan** [apache/druid#9313]
# doubleMean aggregator not available

### Affected Version
version: 0.17.0
"signle  server" mode start
./bin/start-micro-quickstart

### Description
- Cluster size： signle server    ./bin/start-micro-quickstart
- Configurations in use: default configuration
-  Steps to reproduce the problem
1. open web query console
2. enter the following statement and run
```
{
    "queryType": "groupBy",
    "dataSource": "bi-test-two",
    "granularity": "all",
    "dimensions": [
        {
            "type" : "default",
            "dimension" : "业务员名称",
            "outputName": "业务员名称2",
            "outputType": "STRING"
        },
        {
            "type": "extraction",
            "dimension": "单据日期2",
            "outputName": "单据日期2",
            "extractionFn" : {
                "type" : "timeFormat",
                "format" : "yyyy"
            }
        }
    ],
    "limitSpec": {
        "type": "default",
        "limit": 5000,
        "columns": [
            "业务员名称2"
        ]
    },
    "filter": null,
    "aggregations": [
        { "type" : "doubleMean", "name" : "sum(利润)", "fieldName" : "利润"},
        { "type" : "count", "name" : "count$"}
    ],
    "postAggregations": null,
    "intervals": [
        "1990-01-01T00:00:00.000/2112-01-03T00:00:00.000"
    ]
}
```
- error message:
```
Unknown exception / java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.druid.java.util.common.IAE: Unknown object type [[B] / java.lang.RuntimeException / on host localhost:8083
```

- log error message
```
2020-02-05T09:00:25,840 WARN [ForkJoinPool-1-worker-3] org.apache.druid.client.JsonParserIterator - Query [eb33a226-7b8f-41c1-8556-2ba9ff59636f] to host [localhost:8083] interrupted
org.apache.druid.query.QueryInterruptedException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.druid.java.util.common.IAE: Unknown object type [[B]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_242]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_242]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_242]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_242]
	at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call(AnnotatedConstructor.java:124) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:283) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.ValueInstantiator.createFromObjectWith(ValueInstantiator.java:229) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:198) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:422) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.std.ThrowableDeserializer.deserializeFromObject(ThrowableDeserializer.java:65) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4173) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2467) ~[jackson-databind-2.10.1.jar:2.10.1]
	at org.apache.druid.client.JsonParserIterator.init(JsonParserIterator.java:165) [druid-server-0.17.0.jar:0.17.0]
	at org.apache.druid.client.JsonParserIterator.hasNext(JsonParserIterator.java:95) [druid-server-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.BaseSequence.makeYielder(BaseSequence.java:89) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.BaseSequence.toYielder(BaseSequence.java:69) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.MappedSequence.toYielder(MappedSequence.java:49) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$ResultBatch.fromSequence(ParallelMergeCombiningSequence.java:847) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$SequenceBatcher.block(ParallelMergeCombiningSequence.java:897) [druid-core-0.17.0.jar:0.17.0]
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3313) [?:1.8.0_242]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$SequenceBatcher.getBatchYielder(ParallelMergeCombiningSequence.java:886) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$YielderBatchedResultsCursor.initialize(ParallelMergeCombiningSequence.java:993) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$PrepareMergeCombineInputsAction.compute(ParallelMergeCombiningSequence.java:702) [druid-core-0.17.0.jar:0.17.0]
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [?:1.8.0_242]
2020-02-05T09:00:25,847 WARN [qtp837981622-121[groupBy_[bi-test-two]_eb33a226-7b8f-41c1-8556-2ba9ff59636f]] org.apache.druid.server.QueryLifecycle - Exception while processing queryId [eb33a226-7b8f-41c1-8556-2ba9ff59636f] (QueryInterruptedException{msg=java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.druid.java.util.common.IAE: Unknown object type [[B], code=Unknown exception, class=java.lang.RuntimeException, host=localhost:8083})
2020-02-05T09:03:20,879 WARN [ForkJoinPool-1-worker-5] org.apache.druid.client.JsonParserIterator - Query [e4929ff5-703b-4ad7-b706-fb0063328a9a] to host [localhost:8083] interrupted
org.apache.druid.query.QueryInterruptedException: java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.druid.java.util.common.IAE: Unknown object type [[B]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[?:1.8.0_242]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[?:1.8.0_242]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[?:1.8.0_242]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423) ~[?:1.8.0_242]
	at com.fasterxml.jackson.databind.introspect.AnnotatedConstructor.call(AnnotatedConstructor.java:124) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.std.StdValueInstantiator.createFromObjectWith(StdValueInstantiator.java:283) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.ValueInstantiator.createFromObjectWith(ValueInstantiator.java:229) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:198) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:422) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.std.ThrowableDeserializer.deserializeFromObject(ThrowableDeserializer.java:65) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:159) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.ObjectMapper._readValue(ObjectMapper.java:4173) ~[jackson-databind-2.10.1.jar:2.10.1]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2467) ~[jackson-databind-2.10.1.jar:2.10.1]
	at org.apache.druid.client.JsonParserIterator.init(JsonParserIterator.java:165) [druid-server-0.17.0.jar:0.17.0]
	at org.apache.druid.client.JsonParserIterator.hasNext(JsonParserIterator.java:95) [druid-server-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.BaseSequence.makeYielder(BaseSequence.java:89) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.BaseSequence.toYielder(BaseSequence.java:69) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.MappedSequence.toYielder(MappedSequence.java:49) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$ResultBatch.fromSequence(ParallelMergeCombiningSequence.java:847) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$SequenceBatcher.block(ParallelMergeCombiningSequence.java:897) [druid-core-0.17.0.jar:0.17.0]
	at java.util.concurrent.ForkJoinPool.managedBlock(ForkJoinPool.java:3313) [?:1.8.0_242]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$SequenceBatcher.getBatchYielder(ParallelMergeCombiningSequence.java:886) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$YielderBatchedResultsCursor.initialize(ParallelMergeCombiningSequence.java:993) [druid-core-0.17.0.jar:0.17.0]
	at org.apache.druid.java.util.common.guava.ParallelMergeCombiningSequence$PrepareMergeCombineInputsAction.compute(ParallelMergeCombiningSequence.java:702) [druid-core-0.17.0.jar:0.17.0]
	at java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:189) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692) [?:1.8.0_242]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [?:1.8.0_242]
2020-02-05T09:03:20,880 WARN [qtp837981622-133[groupBy_[bi-test-two]_e4929ff5-703b-4ad7-b706-fb0063328a9a]] org.apache.druid.server.QueryLifecycle - Exception while processing queryId [e4929ff5-703b-4ad7-b706-fb0063328a9a] (QueryInterruptedException{msg=java.util.concurrent.ExecutionException: java.lang.RuntimeException: org.apache.druid.java.util.common.IAE: Unknown object type [[B], code=Unknown exception, class=java.lang.RuntimeException, host=localhost:8083})

```

**Comment by suneet-s** [comment]
This appears to have been fixed by #9568 which was released in 0.18

@BeanNan Can you confirm that the doubleMean aggregator works for you in this version?

**Comment by anagao** [comment]
I had the same issue, I can confirm upgrading to 0.18 resolved this for me

**Comment by suneet-s** [comment]
Thanks for confirming @anagao ! Closing as fixed by #9568
